plugins/module_utils/openshift_adm_prune_deployments.py

```diff
@@ -0,0 +1,139 @@
+
+        def _zeroReplicaSize(obj):
+            return obj['spec']['replicas'] == 0 and obj['status']['replicas'] == 0
+
+        def _complete_failed(obj):
+            DeploymentStatusAnnotation = "openshift.io/deployment.phase"
+            try:
+                # validate that replication controller status is either 'Complete' or 'Failed'
+                deployment_phase = obj['metadata']['annotations'].get(DeploymentStatusAnnotation)
+                return deployment_phase in ('Failed', 'Complete')
+            except Exception:
+                return False
+
+        def _younger(obj):
+            creation_timestamp = datetime.strptime(obj['metadata']['creationTimestamp'], '%Y-%m-%dT%H:%M:%SZ')
+            now = datetime.now(timezone.utc).replace(tzinfo=None)
+            age = (now - creation_timestamp).seconds / 60
+            return age > self.params['keep_younger_than']
+
+        def _orphan(obj):
+            try:
+                # verify if the deploymentconfig associated to the replication controller is still existing
+                deploymentconfig_name = get_deploymentconfig_for_replicationcontroller(obj)
+                exists = self.k8s_ansible_mixin.kubernetes_facts(kind="DeploymentConfig",
+                                                                 api_version="apps.openshift.io/v1",
+                                                                 name=deploymentconfig_name,
+                                                                 namespace=self.params.get('namespace'))
+                return not (exists.get['api_found'] and len(exists['resources']) > 0)
+            except Exception:
+                return False
+
+        predicates = [_deployment, _zeroReplicaSize, _complete_failed]
+        if self.params['orphans']:
+            predicates.append(_orphan)
+        if self.params['keep_younger_than']:
+            predicates.append(_younger)
+
+        results = []
+        for item in replicacontrollers:
+            match = True
+            for pred in predicates:
+                if not pred(item):
+                    match = False
+                    break
+            if match:
+                results.append(item)
+        return results
+
+    def execute(self):
+        # list replicationcontroller candidate for pruning
+        kind = 'ReplicationController'
+        api_version = 'v1'
+        api = self.k8s_ansible_mixin.find_resource(kind=kind, api_version=api_version, fail=True)
+        replication_list = api.get()
+        candidates = self.filter_replication_controller(replication_list.items)
+
+        if len(candidates) == 0:
+            self.exit_json(changed=False, msg="No candidate ReplicationController for pruning.")
+
+        candidates = [(rep['metadata']['namespace'], rep['metadata']['name']) for rep in candidates]
+        if self.check_mode:
+            self.exit_json(changed=False, replication_controllers=[x + "/" + y for x, y in candidates])
```

Comment:
Shouldn't this be `changed=True`?

Reply:
I think I was wrong on how the ``check_mode`` should work.
Thanks